android/src/main/java/com/facebook/reactnative/androidsdk/FBProfileModule.java

```diff
@@ -18,24 +18,33 @@
     public static final String NAME = "FBProfile";
 
     public FBProfileModule(ReactApplicationContext reactContext) {
-      super(reactContext);
+        super(reactContext);
     }
 
     @NonNull
     @Override
     public String getName() {
-    return NAME;
-  }
+        return NAME;
+    }
 
     /**
-    * Get the current logged profile.
-    * @param callback Use callback to pass the current logged profile back to JS.
-    */
+     * Get the current logged profile.
+     *
+     * @param callback Use callback to pass the current logged profile back to JS.
+     */
     @ReactMethod
     public void getCurrentProfile(Callback callback) {
         //Return the profile object as a ReactMap.
-        callback.invoke(Profile.getCurrentProfile() == null
-                ? null
-                : Utility.profileToReactMap(Profile.getCurrentProfile()));
-  }
+        if (Profile.getCurrentProfile() == null) {
+            mProfileTracker = new ProfileTracker() {
```

Comment:
it doesn't appear that you have actually tested this at all? I say that because this won't compile, and that's a worrying sign - https://github.com/thebergamo/react-native-fbsdk-next/runs/5838884802?check_suite_focus=true#step:9:395